File: CS/WebApplication1/CustomControls/CustomAppointmentFormController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.XtraScheduler;

namespace WebApplication1.CustomControls {
    public class CustomAppointmentFormController : AppointmentFormController {
        public CustomAppointmentFormController(ASPxScheduler control, Appointment apt) : base(control, apt) {
        
        }

        public int CompanyIDField {
            get { return Convert.ToInt32(EditedAppointmentCopy.CustomFields["AppointmentCompany"]); }
            set { EditedAppointmentCopy.CustomFields["AppointmentCompany"] = value; }
        }

        public int ContactIDField {
            get { return Convert.ToInt32(EditedAppointmentCopy.CustomFields["AppointmentContact"]); }
            set { EditedAppointmentCopy.CustomFields["AppointmentContact"] = value; }
        }
    }
}

File: VB/WebApplication1/CustomControls/CustomAppointmentFormController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.XtraScheduler

Namespace WebApplication1.CustomControls
	Public Class CustomAppointmentFormController
		Inherits AppointmentFormController
		Public Sub New(ByVal control As ASPxScheduler, ByVal apt As Appointment)
			MyBase.New(control, apt)

		End Sub

		Public Property CompanyIDField() As Integer
			Get
				Return Convert.ToInt32(EditedAppointmentCopy.CustomFields("AppointmentCompany"))
			End Get
			Set(ByVal value As Integer)
				EditedAppointmentCopy.CustomFields("AppointmentCompany") = value
			End Set
		End Property

		Public Property ContactIDField() As Integer
			Get
				Return Convert.ToInt32(EditedAppointmentCopy.CustomFields("AppointmentContact"))
			End Get
			Set(ByVal value As Integer)
				EditedAppointmentCopy.CustomFields("AppointmentContact") = value
			End Set
		End Property
	End Class
End Namespace